experimental/list_formatter/src/provider.rs

```diff
@@ -75,6 +75,22 @@ impl<'data> ListFormatterPatternsV1<'data> {
     pub fn pair(&'data self, width: Width) -> &ConditionalListJoinerPattern<'data> {
         &self.0[4 * (width as usize) + 3]
     }
+
+    /// The expected number of bytes required by the list literals to join a list of length
+    /// len. If none of the patterns are conditional, this is exact, otherwise it is an
+    /// upper bound.
+    pub fn size_hint(&'data self, width: Width, len: usize) -> usize {
```

Comment:
Nit, here and above: you should not have `'data` in the `self` argument; it should be just `&self`

(this is a pre-existing issue)